homeassistant/components/template/button.py

```diff
@@ -0,0 +1,157 @@
+    """Create the Template button."""
+    entities = []
+    for definition in definitions:
+        unique_id = definition.get(CONF_UNIQUE_ID)
+        if unique_id and unique_id_prefix:
+            unique_id = f"{unique_id_prefix}-{unique_id}"
+        entities.append(
+            TemplateButtonEntity(
+                hass,
+                definition[CONF_NAME],
+                definition.get(CONF_AVAILABILITY),
+                definition[CONF_PRESS],
+                definition.get(CONF_DEVICE_CLASS),
+                unique_id,
+                definition.get(CONF_ICON),
+            )
+        )
+    return entities
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: Config,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: dict[str, Any] | None = None,
+) -> None:
+    """Set up the template button."""
+    if discovery_info is None:
+        _LOGGER.warning(
+            "Template button entities can only be configured under template:"
+        )
+        return
+
+    if "coordinator" in discovery_info:
+        async_add_entities(
+            TriggerButtonEntity(hass, discovery_info["coordinator"], config)
+            for config in discovery_info["entities"]
+        )
+        return
+
+    async_add_entities(
+        await _async_create_entities(
+            hass, discovery_info["entities"], discovery_info["unique_id"]
+        )
+    )
+
+
+class TemplateButtonEntity(TemplateEntity, ButtonEntity):
+    """Representation of a template button."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name_template: Template,
+        availability_template: Template | None,
+        command_press: dict[str, Any],
+        device_class: ButtonDeviceClass | None,
+        unique_id: str | None,
+        icon_template: Template | None,
+    ) -> None:
+        """Initialize the button."""
+        super().__init__(
+            availability_template=availability_template, icon_template=icon_template
+        )
+        self._attr_name = DEFAULT_NAME
+        self._name_template = name_template
+        name_template.hass = hass
+        with contextlib.suppress(TemplateError):
+            self._attr_name = name_template.async_render(parse_result=False)
+        self._command_press = Script(hass, command_press, self._attr_name, DOMAIN)
+        self._attr_device_class = device_class
+        self._attr_unique_id = unique_id
+        self._attr_state = None
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        if self._name_template and not self._name_template.is_static:
+            self.add_template_attribute("_attr_name", self._name_template, cv.string)
+        await super().async_added_to_hass()
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        await self._command_press.async_run(context=self._context)
+
+
+class TriggerButtonEntity(TriggerEntity, ButtonEntity):
```

Comment:
How is this useful, is there some reasonable use case?

Reply:
tbh I originally was not planning to include it, but the effort was low. I do not have a good use case for it so maybe it's best to remove it.